onetl/connection/db_connection/kafka.py

```diff
@@ -0,0 +1,197 @@
+#  Copyright 2023 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import logging
+import os
+from typing import TYPE_CHECKING, Any, ClassVar, List, Optional
+
+from etl_entities.instance import Cluster
+from pydantic import SecretStr, root_validator, validator
+
+from onetl.connection.db_connection.db_connection import DBConnection
+from onetl.hwm import Statement
+from onetl.impl import LocalPath
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+    from pyspark.sql.types import StructType
+
+PARTITION_OVERWRITE_MODE_PARAM = "spark.sql.sources.partitionOverwriteMode"
+log = logging.getLogger(__name__)
+
+
+class Kafka(DBConnection):
+    """
+    This connector is designed to read and write from Kafka in batch mode.
+
+    .. dropdown:: Version compatibility
+
+        * Apache Kafka versions: 0.10 or higher
+        * Spark versions: 2.0.x - 3.3.x
+
+    Parameters
+    ----------
+
+    addresses : list[str]
+        A list of broker addresses, for example [192.168.1.10:9092, 192.168.1.11:9092].
+        The list cannot be empty.
+
+    cluster : Cluster
+        Cluster name. Used for HWM and lineage. A cluster field cannot be empty.
+
+    user : str, default  ``None``
+        User, which have proper access to the Apache Kafka. For example: ``some_user``.
+
+    password : SecretStr, default  ``None``
+        Password for Kafka connection.
+
+    keytab : LocalPath, default  ``None``
+        A path to the keytab file. A keytab is a file containing pairs of Kerberos principals and encrypted keys that
+        are derived from the Kerberos password. You can use this file to log on to Kerberos without being prompted for
+        a password.
+
+    Examples
+    --------
+
+    .. code:: python
+        kafka = Kafka(
```

Comment:
```suggestion
    .. code:: python

        kafka = Kafka(
```

Без перевода строки в документации не будет подсветки синтаксиса. Ниже тоже добавь